src/components/network/line-layer.js

```diff
@@ -195,336 +206,808 @@ class LineLayer extends CompositeLayer {
+                        getLineAngles: (arrow) => [
+                            arrow.line.geo.angleStart,
+                            arrow.line.geo.angle,
+                            arrow.line.geo.angleEnd,
+                        ],
+                        getProximityFactors: (arrow) => [
+                            arrow.line.geo.proximityFactorStart,
+                            arrow.line.geo.proximityFactorEnd,
+                        ],
+                        getDistanceBetweenLines: props.distanceBetweenLines,
+                        maxParallelOffset: props.maxParallelOffset,
+                        minParallelOffset: props.minParallelOffset,
+                        getDirection: (arrow) => {
+                            return getArrowDirection(arrow.line.p1);
+                        },
+                        animated:
+                            props.showLineFlow &&
+                            props.lineFlowMode === LineFlowMode.ANIMATED_ARROWS,
+                        visible:
+                            props.showLineFlow &&
+                            props.filteredNominalVoltages.includes(
+                                nominalVoltage
+                            ) &&
+                            state[PATH_UPDATED + nominalVoltageIndex] <=
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                        updateTriggers: {
+                            getLinePositions: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getLineParallelIndex: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getLineAngles: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getColor: [
+                                props.lineFlowColorMode,
+                                props.lineFlowAlertThreshold,
+                                props.updatedLines,
+                            ],
+                        },
+                    })
+                )
+        );
+        this.setState({ layers: layers });
+        let next = undefined;
+        if (
+            doNextNominalVoltage &&
+            this.hasNextNominalVoltage(nominalVoltageIndex)
+        )
+            next = (ts) =>
+                this.computeArrows(
+                    ts,
+                    nominalVoltageIndex + 1,
+                    doLabel,
+                    doNextNominalVoltage
                 );
-                line.proximityFactorEnd = this.getProximityFactor(
-                    positions[positions.length - 2],
-                    positions[positions.length - 1]
+        else if (doLabel)
+            next = (ts) => this.computeLabels(ts, 0, doNextNominalVoltage);
+        if (next !== undefined) this.scheduleNext(timestamp, next);
```

Comment:
scheduleNext already does the if undefined, remove it ?

Reply:
Or does scheduleNext really need to do the check ? In this case, it's better to remove it in scheduleNext ?